tests/unit/util/main.c

```diff
@@ -867,6 +867,28 @@ ZTEST(util, test_utf8_lcpy_truncated)
 	zassert_str_equal(dest_str, expected_result, "Failed to copy");
 }
 
+ZTEST(util, test_utf8_is_valid)
+{
+    /* Test whether the verification function meets the requirements */
+	zassert_true(utf8_is_valid("\xce\xba\xe1\xbd\xb9\xcf\x83\xce\xbc\xce\xb5", 11));
+	zassert_true(utf8_is_valid("\x00", 1));
```

Comment:
Would be very helpful to know what a test is actually doing.
```suggestion
	zassert_true(utf8_is_valid("\x00", 1), "Describe what this test does");
```

Reply:
I edited [this message](https://github.com/zephyrproject-rtos/zephyr/pull/80779#discussion_r1826984496) to help with that.

Reply:
We can also use the actual UTF-8 string instead. 

e.g. `zassert_true(utf8_is_valid("\xc2\x80", 2));` could be something like `zassert_true(utf8_is_valid("€", sizeof("€"));`

That makes these much easier to read, and more clear that they are testing valid UTF-8 strings. 

When I tried to decode `\xc2\x80` (which may be a valid UTF-8), I couldn't get any character. 

The `zassert_false` is a bit more tricky

Reply:
The test is not visible to the user and I don't think it needs to be that readable. It exists only to verify that the function works, nothing more.

Reply:
I would disagree. The tests should be readable and understandable by other developers. It is currently not IMO since we have reviewers commenting on this. 

I am not a computer, so I have no idea if e.g. `\xed\x9f\xbf` is in fact a valid UTF8 string, and thus cannot determine if the check here is correct or not. I could try and parse each individual line here manually, but that is a waste of time, when it's easier to add a description, a comment or use UTF8 characters like "Øåæ".

Reply:
Most of the tests provided by @josuah  do not have an explicit string and are only used to test whether the decoder is comprehensive.

Reply:
Please add those comments in a sensible way. It will help people to understand the code you create here.